pkg/rbd/rbd_journal.go

```diff
@@ -0,0 +1,301 @@
+	}
+
+	return nil
+}
+
+func validateRbdSnap(rbdSnap *rbdSnapshot) error {
+	if err := validateNonEmptyField(rbdSnap.RequestName, "RequestName", "rbdSnapshot"); err != nil {
+		return err
+	}
+
+	if err := validateNonEmptyField(rbdSnap.Monitors, "Monitors", "rbdSnapshot"); err != nil {
+		return err
+	}
+
+	if err := validateNonEmptyField(rbdSnap.AdminID, "AdminID", "rbdSnapshot"); err != nil {
+		return err
+	}
+
+	if err := validateNonEmptyField(rbdSnap.Pool, "Pool", "rbdSnapshot"); err != nil {
+		return err
+	}
+
+	if err := validateNonEmptyField(rbdSnap.RbdImageName, "RbdImageName", "rbdSnapshot"); err != nil {
+		return err
+	}
+
+	if err := validateNonEmptyField(rbdSnap.ClusterID, "ClusterID", "rbdSnapshot"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateRbdVol(rbdVol *rbdVolume) error {
+	if err := validateNonEmptyField(rbdVol.RequestName, "RequestName", "rbdVolume"); err != nil {
+		return err
+	}
+
+	if err := validateNonEmptyField(rbdVol.Monitors, "Monitors", "rbdVolume"); err != nil {
+		return err
+	}
+
+	if err := validateNonEmptyField(rbdVol.AdminID, "AdminID", "rbdVolume"); err != nil {
+		return err
+	}
+
+	if err := validateNonEmptyField(rbdVol.Pool, "Pool", "rbdVolume"); err != nil {
+		return err
+	}
+
+	if err := validateNonEmptyField(rbdVol.ClusterID, "ClusterID", "rbdVolume"); err != nil {
+		return err
+	}
+
+	if rbdVol.VolSize == 0 {
+		return errors.New("value 'VolSize' in 'rbdVolume' structure cannot be 0")
+	}
+
+	return nil
+}
+
+/*
+checkSnapExists, and its counterpart checkVolExists, function as checks to determine if passed
+in rbdSnapshot or rbdVolume exists on the backend.
+
+**NOTE:** These functions manipulate the rados omaps that hold information regarding
+volume names as requested by the CSI drivers. Hence, these need to be invoked only when the
+respective CSI driver generated snapshot or volume name based locks are held, as otherwise racy
+access to these omaps may end up leaving them in an inconsistent state.
+
+These functions need enough information about cluster and pool (ie, Monitors, Pool, IDs filled in)
+to operate. They further require that the RequestName element of the structure have a valid value
+to operate on and determine if the said RequestName already exists on the backend.
+
+These functions populate the snapshot or the image name, its attributes and the CSI snapshot/volume
+ID for the same when succesful.
+
+These functions also cleanup omap reservations that are stale. I.e when omap entries exist and
+backing images or snapshots are missing, or one of the omaps exist and the next is missing. This is
+because, the order of omap creation and deletion are inverse of each other, and protected by the
+request name lock, and hence any stale omaps are leftovers from incomplete transactions and are
+hence safe to garbage collect.
+*/
+func checkSnapExists(rbdSnap *rbdSnapshot, credentials map[string]string) (found bool, err error) {
```

Comment:
found is not used  in this  function

Reply:
Done (removed any further instances of naming return vars as well)